src/app/app-design-tool/components/jobs-list/jobs-list.component.html

```diff
@@ -0,0 +1,8 @@
+<mat-card>
+  <mat-card-content>
+    <mat-list *ngFor="let job of jobs">
```

Comment:
Here you're actually creating many `mat-list` items. Check the generated HTML. You should put `*ngFor` on `app-job`. The divider is tricky since you want that between elements but this way you'll also add it to the last element.